File: src/Contexts/Ordering/Domain/Entities/Buyer/Entities/Address/Address.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace eShop.Ordering.Buyer.Entities.Address
{
    public class Address : Aggregates.Entity<Address, State, Buyer>
    {
        private Address() { }

        public void Add(string alias, string street, string city, string state, string country, string zipCode)
        {
            Apply<Events.Added>(x =>
            {
                x.UserName = Parent.Id;
                x.AddressId = Id;
                x.Alias = alias;
                x.Street = street;
                x.City = city;
                x.State = state;
                x.Country = country;
                x.ZipCode = zipCode;
            });
        }

        public void Remove()
        {
            Apply<Events.Removed>(x =>
            {
                x.UserName = Parent.Id;
                x.AddressId = Id;
            });
        }
    }
}

File: src/Web/src/app/models/ordering/address.ts
import { types } from 'mobx-state-tree';

export interface AddressType {
  id: string;

  alias: string;
  street: string;
  city: string;
  state: string;
  country: string;
  zipCode: string;
}
export const AddressModel = types
  .model('Ordering_Buyer_Address', {
    id: types.identifier(types.string),

    alias: types.string,
    street: types.string,
    city: types.string,
    state: types.string,
    country: types.string,
    zipCode: types.string
  });
